ocnovi_oop2.rb

```diff
@@ -0,0 +1,129 @@
+class Station # класс станция
+  attr_reader :trains, :name # возвращает список поездов, находящися на станции
+
+  def initialize(name_station) # имя станции указывается при создании
+    @name = name_station
+    @trains = [] # массив поездов
+  end
+
+  def add_train(train) # принимает поезда по одному за раз
+    @trains << train
+  end
+
+  def delete_train(train) # отправляет поезда по одному за раз
+    @trains.delete(train)
+  end
+
+  def type_trains(type) # возвращает список поездов типа type
+    self.trains.find_all { |train| train.type == type }
+  end
+
+  def type_train_numder(type) # возвращает количество поездов данного типа
+    self.trains.count { |train| train.type == type }
+  end
+
+end
+
+class Route # класс маршрут
+  attr_reader :stations # возвращает список всех станций маршрута
+
+  def initialize(start, finish) # начальная и конечная станция указываются при создании
+    @stations = [start, finish] #массив станций
+  end
+
+  def first_station # первая станция
+    @stations[0]
+  end
+
+  def last_station # последняя станция
+    @stations[-1]
+  end
+
+  def print # выводит список всех станций по-порядку от начальной до конечной
+    stations.each{ |st| puts st.name }
+  end
+
+  def add_station(station) # добавлении станции в маршрут
+    stations.insert(-2,station)
+  end
+
+  def delete_station(station) # удалении станции из маршрута
+    if station != @stations[0] && station != @stations[-1]
```

Comment:
у тебя есть же методы (см. выше)
в таком варианте легче читать код:
```suggestion
    if station != first_station && station != last_station
```